serde/src/de/impls.rs

```diff
@@ -66,13 +68,37 @@ impl<'de> Visitor<'de> for BoolVisitor {
     }
 }
 
+struct BoolFromVisitor<'a>(&'a mut bool);
```

Comment:
I don't think primitive types need this specialization. The pointer indirection will probably be less performant than the default implementation. I'm easily convinced otherwise by benchmarks ;)

Reply:
It's been a while since I implemented this, but I seem to recall getting wins from doing this. Gonna start profiling tomorrow, will find number then.